### DISCUSSION TITLE
RandomIC for different boundaries or blocks

### DISCUSSION THREAD

**Discussion by bylore** [idaholab/moose#28720] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi there.
I am using RandomIC to set the initial condition for my variable 'eta'. When i don't specify the block or the boundary for this random initial conditions, i get the right setting.
![image](https://github.com/user-attachments/assets/ab575901-f4ed-457c-84df-618a05e5ae96)
When i specify the blocks, i get the same right results as above.
But when i specify the boundaries instead of the blocks, i get the following.
![image](https://github.com/user-attachments/assets/7091c46a-0e44-4b87-b215-10ed3e624ea9)
I use 25 processors for this simulation. It seems that RandomIC only assign random values for every processor in this case.
Here is my script

```
[body_eta_random]
    type = RandomIC
    variable = eta
    #boundary = "b1_break b2_defect"
    block = "b1_break_QUAD4 b1_break_TRI3 b2_defect_QUAD4"
    max = 1.0
    min = 0.9
    seed = 0
  []
```
My MOOSE version is git commit 42342ed08d on 2023-05-18.
Could you give me some clue?

**Comment by GiudGiud** [comment]
Hello

The IC does not apply on the interior of the blocks when it is applied on the boundaries. It only applies to the single element that lies on the interior side of the boundary.

Now as to why each process has a different value anyway, I m not sure. 
What are you trying to do here? Block restrict the randomness to a single region?

**Reply by bylore** [reply]
Yeah, I want restrict the randomness to some specific region. At first, I use the parameter `block` to indict the block where some value I want assign. But I found each node on the boundary of two blocks has two block id number (corresponding to these two blocks). So sometimes I can't assign the right value I want to this kind of nodes. Later I found that making the same set of nodes in one `block` into `boundary` and use the parameter `boundary` can avoid the "two-block-id" problem.  This works fine for ConstanIC. But for RandomIC, I got the issue discribled above.

**Reply by bylore** [reply]
`boundary` and `block` here represents same set of nodes, but I got different initial values. I'm reporting this to see if I'm using the `boundary` parameter incorrectly or if this is a potential bug.

**Reply by GiudGiud** [reply]
the ICs will likely override each other on those nodes. you have 3 ICs setting a value on the same node

**Reply by GiudGiud** [reply]
are you using multiple ICs then?

**Reply by bylore** [reply]
No, in the issue case, only one IC was used for the concerned variable. I just tied again.
Using
>   [rand_random]
    type = RandomIC
    variable = rand
    #boundary = "b1_break b2_nofail b3_defect"
    block = "b1_break_QUAD4 b1_break_TRI3 b2_nofail_QUAD4 b2_nofail_TRI3 b3_defect_QUAD4"
    max = 1.0
    min = 0.9
    seed = 0
  []

> 

I got 
![image](https://github.com/user-attachments/assets/50424da2-b08a-435a-804c-78a2fc41cdd8)

While using
>   [rand_random]
    type = RandomIC
    variable = rand
    boundary = "b1_break b2_nofail b3_defect"
    #block = "b1_break_QUAD4 b1_break_TRI3 b2_nofail_QUAD4 b2_nofail_TRI3 b3_defect_QUAD4"
    max = 1.0
    min = 0.9
    seed = 0
  []
I got
![image](https://github.com/user-attachments/assets/b2140869-7903-407d-937c-8246c3386ee2)
This time 10 processors were used.

**Reply by GiudGiud** [reply]
does it improve if you specify a distribution?

I ll raise an issue for this problem. But using "boundary' does not fix your problem anyway so I recommend you try other ways to get the IC you want. 
Probably a single IC with some logic to work on nodes that are shared between multiple blocks

**Reply by bylore** [reply]
Thank you. I can avoid this issue so it does not affect my use.  In my offline installed moose, there is no registered distribution object currently. I‘ll get back to you if I have these results later.
